meshnets/modules/lightning_wrapper.py

```diff
@@ -8,26 +8,73 @@
 class MGNLightningWrapper(pl.LightningModule):
     """Simple pytorch_lightning wrapper for the MeshGraphNet training logic."""
 
-    def __init__(self, model: torch.nn.Module, learning_rate: float = 1e-3):
+    def __init__(self,
+                 model: torch.nn.Module,
+                 data_stats: dict,
+                 learning_rate: float = 1e-3):
+        """Initialize for a given model, a dictionnary of
+        data statistics for normalization, and a learning rate.
+        
+        The statistics dictionnary is expected to have
+        the following fields:
+            'x_mean'
+            'x_std'
+            'edge_attr_mean'
+            'eadge_attr_std'
+            'y_mean'
+            'y_std'
+        """
         super().__init__()
+
         self.model = model
+        self.data_stats = data_stats
         self.learning_rate = learning_rate
 
+    def normalize_input(self, batch: Batch) -> Batch:
+        """Normalize the input features in a batch.
+        
+        The node and edge features are normalized using the mean and std
+        statistics found in the data_stats dictionnary."""
+
+        # Send the stats to the same device as the data and normalize
+        batch.x = (batch.x - self.data_stats['x_mean'].to(
+            batch.x.device)) / self.data_stats['x_std'].to(batch.x.device)
+
```

Comment:
Move to the GPU on __init__ in order to perform this just once

Reply:
I think this implementation is the better option:
- at init time, we do not yet have information about where lightning is going to put the data (cpu/gpu) so we would need to 'hardcode' that using the config parameters
- if we train on multiple GPUs (e.g. tuning), I believe we would need to send the data to the same GPU as the batch during training, not any GPU

I timed the operation and it takes 2.5s for 100'000 repetitions, which I think is negligible.

Reply:
Good points